### ISSUE TITLE
"Test type arguments in operator ==(Object other)" is a very confusing error message

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1398]
_From @Hixie on December 30, 2018 19:28_

This code:
```dart
  @override
  bool operator ==(dynamic other) {
    if (other.runtimeType != runtimeType)
      return false;
    final AvatarImage typedOther = other as AvatarImage; // line 465
    return username == typedOther.username
        && photoManager == typedOther.photoManager
        && twitarr == typedOther.twitarr;
  }
```
Led to this message from the analyzer:
```
[info] Test type arguments in operator ==(Object other) (/home/ianh/dev/cruisemonkey/lib/src/logic/cruise.dart:465:36)
```

I have no idea what this means.

_Copied from original issue: dart-lang/sdk#35522_

**Comment by pq** [comment]
_From @Hixie on December 30, 2018 19:28_

(See http://dart-lang.github.io/linter/lints/test_types_in_equals.html for what it actually means.)

**Comment by pq** [comment]
_From @Hixie on December 30, 2018 19:30_

See also https://github.com/dart-lang/sdk/issues/35523 which is about the lint itself being bad. This specific issue is about the wording of the lint being unclear.
